### PR TITLE
assignment1-haojin

### PR DESCRIPTION
**Pull request by haojinliang**
assignment 1 haojin

### CODE REVIEW

**Review by stargrep — CHANGES_REQUESTED** [review]

File: week02/assignment/assignment1-Haojin.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+assert print_sum(1) == "1"
+assert print_sum(3) == "6"
+assert print_sum(5) == "15"
+
+
+# Q3. Write a program to check is a year is leap year (x is always > 0)
+
+def is_leap_year(year: int) -> bool:
+    if year > 0:
```
[the remark this answers is not in the document]
Comment (haojinliang) [reply week02/assignment/assignment1-Haojin.py:47]: def is_leap_year(year: int) -> bool:
    return True if (year % 400 == 0) or (year % 4 == 0 and year % 100 != 0) else False

File: week02/assignment/assignment1-Haojin.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+assert to_upper_case(["abc", "de"]) == ["ABC", "DE"]
+assert to_upper_case(["Amazon", "Apple"]) == ["AMAZON", "APPLE"]
+
+
+# Q5. Write a program to use only 'and' and 'or' to implement 'xor'
+# https://baike.baidu.com/item/%E5%BC%82%E6%88%96/10993677?fromtitle=xor&fromid=64178
+
+def xor(a: bool, b: bool) -> bool:
+    if bool(a or b) != bool(a and b):
```
[the remark this answers is not in the document]
Comment (haojinliang) [reply week02/assignment/assignment1-Haojin.py:82]: def xor(a: bool, b: bool) -> bool:
    return True if a != b else False

**Review by haojinliang — COMMENTED** [review]
resolved

File: week02/assignment/assignment1-Haojin.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+assert to_upper_case(["abc", "de"]) == ["ABC", "DE"]
+assert to_upper_case(["Amazon", "Apple"]) == ["AMAZON", "APPLE"]
+
+
+# Q5. Write a program to use only 'and' and 'or' to implement 'xor'
+# https://baike.baidu.com/item/%E5%BC%82%E6%88%96/10993677?fromtitle=xor&fromid=64178
+
+def xor(a: bool, b: bool) -> bool:
+    return True if a != b else False
```
Comment (stargrep) [inline week02/assignment/assignment1-Haojin.py:76]: minor: just return a != b

**Review by stargrep — APPROVED** [review]

**Pull request merged**